presto-phoenix/src/test/java/io/prestosql/plugin/phoenix/TestPhoenixDistributedQueries.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.phoenix;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.tpch.TpchTable;
+import io.prestosql.tests.AbstractTestDistributedQueries;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.Test;
+
+@Test
+public class TestPhoenixDistributedQueries
+        extends AbstractTestDistributedQueries
+{
+    public TestPhoenixDistributedQueries()
+            throws Exception
+    {
+        super(() -> PhoenixQueryRunner.createPhoenixQueryRunner(ImmutableMap.of(), TpchTable.getTables()));
+    }
+
+    @Override
+    protected boolean supportsViews()
+    {
+        return false;
+    }
+
+    @Override
+    public void testRenameTable()
+    {
+        // Phoenix does not support renaming tables
+    }
+
+    @Override
+    public void testRenameColumn()
+    {
+        // Phoenix does not support renaming columns
+    }
+
+    @Test
+    public void testDeleteCompoundPk()
+    {
+        assertUpdate("CREATE TABLE test_delete_compound_pk  (pk1 bigint, pk2 varchar, val1 varchar) WITH (ROWKEYS='pk1,pk2')");
+
+        assertUpdate("INSERT INTO test_delete_compound_pk VALUES (1, 'pkVal1', 'val1')", 1);
+        assertUpdate("INSERT INTO test_delete_compound_pk VALUES (2, 'pkVal2', 'val2')", 1);
+        assertUpdate("INSERT INTO test_delete_compound_pk VALUES (3, 'pkVal3', 'val3')", 1);
+
+        assertUpdate("DELETE FROM test_delete_compound_pk where val1 = 'val2'", 1);
+        assertQuery("SELECT count(*) FROM test_delete_compound_pk where val1 = 'val2'", "SELECT 0");
+
+        assertUpdate("DELETE FROM test_delete_compound_pk where pk1 = 3 AND pk2 = 'pkVal3'", 1);
+        assertQuery("SELECT pk1 from test_delete_compound_pk", "SELECT 1");
+    }
+
+    @Test
+    public void testInsert()
+    {
+        @Language("SQL")
+        String query = "SELECT orderdate, orderkey, totalprice FROM orders";
+
+        assertUpdate("CREATE TABLE test_insert WITH (ROWKEYS='orderkey')AS " + query + " WITH NO DATA", 0);
+        assertQuery("SELECT count(*) FROM test_insert", "SELECT 0");
+
+        assertUpdate("INSERT INTO test_insert " + query, "SELECT count(*) FROM orders");
+
+        assertQuery("SELECT * FROM test_insert", query);
+
+        assertUpdate("INSERT INTO test_insert (orderkey) VALUES (-1)", 1);
```

Comment:
Can you run this insert twice?

Reply:
Running twice is an Upsert in Phoenix.  I added the extra insert, doesn't impact the test.

Reply:
will the second INSERT also return `1` as the number of rows changed?

Reply:
Yes, the second INSERT updates 1 row (since it's a Phoenix Upsert).